pkg/strategy/controller_test.go

```diff
@@ -246,38 +246,94 @@ func TestNormalizeUI(t *testing.T) {
 	}{
 		{
 			j:        &v1.JaegerSpec{},
-			expected: &v1.JaegerSpec{},
+			expected: &v1.JaegerSpec{UI: v1.JaegerUISpec{Options: v1.NewFreeForm(map[string]interface{}{"dependencies": map[string]interface{}{"menuEnabled": false}})}},
 		},
 		{
-			j:        &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es.archive.enabled": "false"})}},
-			expected: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es.archive.enabled": "false"})}},
+			j:        &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Type: "memory"}},
+			expected: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Type: "memory"}},
 		},
 		{
 			j: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es-archive.enabled": "true"})}},
 			expected: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es-archive.enabled": "true"})},
-				UI: v1.JaegerUISpec{Options: v1.NewFreeForm(map[string]interface{}{"archiveEnabled": true})}},
+				UI: v1.JaegerUISpec{Options: v1.NewFreeForm(map[string]interface{}{"archiveEnabled": true, "dependencies": map[string]interface{}{"menuEnabled": false}})}},
```

Comment:
Isn't this wrong - if the storage type is not defined, it should default to `memory`, which should show the dependencies tab. The expected results is showing it disabled.

Reply:
In general yes, but this tests only normalizeUI which works on any storage